### DISCUSSION TITLE
窗口系列参数 补充内容

### DISCUSSION THREAD

**Discussion by hooke007** [hooke007/mpv_PlayKit#69] [INFO 信息]
_补充一些其它的常用窗口类参数。这不是手册的汉化，更多信息和可用参数请阅读原文_

官方手册定位： https://mpv.io/manual/master/#window

- **`--title=<string>`**
边框上的窗口标题，支持监视 [属性](https://mpv.io/manual/master/#property-list) (也支持[属性扩展](https://mpv.io/manual/master/#property-expansion)语法) 并动态刷新。
虽然很酷但是最好不要这么做（监听一个时刻都在变化的选项可能产生性能问题），比如： `title="实时帧率 ${estimated-display-fps}"`
![title](https://user-images.githubusercontent.com/41094733/147036058-ac579244-555d-44c3-bda9-3cb22f694604.jpg)

- **`--fullscreen=<yes|默认no>`**
该参数可简写为 `--fs`
启动时全屏？
🔺全屏状态不属于窗口

- **`--keep-open=<yes|默认no|always>`**
 **yes** 当前文件播完自动切换到下一个文件，直到播放到最后一个文件时暂停在最后一帧
 **no** 不间断播完列表中的文件后，关闭程序
 **always** 永远在当前文件的最后一帧暂停（需要手动切换到下一个文件）

- **`--drag-and-drop=<no|默认auto|replace|append>`**
定义外部文件（主要是音视频类）拖拽进mpv时的行为。
 **no** 完全禁用
 **auto** 遵循系统底层设置，在win10上默认为替换当前的播放文件，拖放时按住 <kbd>shift</kbd> 则为追加到当前播放列表
 **replace** 强制为替换当前的播放文件
 **append** 强制为追加到当前的播放列表

- **`--ontop=<yes|默认no>`**
窗口置顶？
🔺当 `gpu-context=winvk` / `--gpu-api=vulkan` 时，且当切换到全屏时附带有独占全屏的作用

- **`--border=<默认yes|no>`**
窗口边框？
无边框模式的标题与 `title=<string>` 无关。内置的原版OSC标题（与原版默认进度条标题的内容一致）由此参数控制
https://github.com/hooke007/mpv_PlayKit/blob/82dde4fbdf57a486c8527f9934e7585c9ce00d1b/portable_config/script-opts/osc.conf#L53-L55
**osc_plus.lua 的版本实现了该标题的独立设置** <br><img src="https://user-images.githubusercontent.com/41094733/147036098-1a17641b-4c2e-4c28-a3fe-4fe0fc5673a6.jpg" width="400">
🔺无边框模式会失去部分特性，例如边缘贴靠<br>
    - **`--title-bar=<默认yes|no>`**
    这里建议使用这个参数实现视觉无边框（当然也可能存在bug，但至少保留了窗口管理器的特性）

- **`--geometry=<[W[xH]][+-x+-y][/WS]>`** , **`--geometry=<x:y>`**
启动时的具体窗口尺寸（此时具有冻结窗口尺寸的功能）和/或位置（从边角计算到最近的画面边角）。
若与视频比例不同，自动填充黑边。即使拉伸，也尽可能保持设定比例。
赋值有多种表达方式：
1)仅宽度像素数，例如 `800` （维持源比例，只考量宽度到800像素）
2)仅宽度百分比，例如 `80%` （维持源比例，只考量宽度到0.8显示器输出分辨率）
3)宽度和高度百分比，例如 `75%x70%` （初始窗口为显示器输出分辨率的0.75宽和0.7高）
4)宽度和高度像素数，例如 `1500x1000` （初始窗口为1500像素宽度和1000像素高度）
5)在前4种的基础上叠加定位，例如 `75%x70%+40-20` （初始窗口为目标大小，并放置在左下角目标位置）
6)宽度与高度百分比，例如 `40%:30%` （初始窗口左上角位于x轴0.4和y轴0.3处）
7)宽度与高度像素数，例如 `300:200` （初始窗口左上角位于x轴300像素和y轴200像素处）
🔺有边框模式下，百分比的计算结果不准确
🔺使用此项后 `--autofit` 系列和 `--window-scale` 参数将无效

- **`--autofit=<[W[xH]]>`**
自动对当前文件缩放到目标大小
赋值有多种表达方式：
1)仅宽度像素数，例如 `1000` （维持源比例，只考量宽度到1000像素）
2)仅宽度百分比，例如 `75%` （维持源比例，只考量宽度到0.75显示器输出分辨率）
3)宽度和高度百分比，例如 `80%x70%` （尽可能放大窗口但不超过0.8宽度和0.7高度）
4)宽度和高度像素数，例如 `1200x800` （尽可能放大窗口但不超过1200像素宽度和800像素高度）
  - **`--autofit-larger=<[W[xH]]>`**
当缩放分辨率高于显示器输出分辨率时生效：窗口模式下最大尺寸（例如在FHD屏上打开4k视频初始窗口过大）
  - **`--autofit-smaller=<[W[xH]]>`**
当缩放分辨率低于显示器输出分辨率时生效：窗口模式下最小尺寸（例如在4k屏上打开720p视频初始窗口过小）
🔺 有边框模式下，百分比的计算结果不准确
🔺 `autofit-larger` 和 `autofit-smaller` 的优先级高于 `--autofit=<[W[xH]]>`

- **`--window-scale=<factor>`**
自动窗口缩放倍率，默认为1。但是只有在 `hidpi-window-scale=no` 时才能让窗口缩放为真实的1（否则会纳入系统DPI考量）
🔺 `window-scale` 的优先级低于 `geometry` 和 `--autofit` 系列
🔺在状态信息stats的第一页，即使在全屏状态，始终会显示切换到全屏前的窗口缩放系数
  - **`--window-minimized=<yes|默认no>`**
启动时窗口最小化？（这玩意的用途是什么。。。）
  - **`--window-maximized=<yes|默认no>`**
启动时窗口最大化？
🔺无边框模式下的最大化（可能无视并覆盖任务栏）在视觉上和全屏状态相似，但它仍属于窗口状态

- **`--keepaspect=<默认yes|no>`**
 **yes** 按源比例拉伸视频
 **no** 变形画面至窗口形状
  - **`--keepaspect-window=<默认yes|no>`**
   **yes** 窗口比例锁定为视频比例
   **no** 窗口自由拉伸（当 `keepaspect=yes` 时四周填充黑边）

- **`--hidpi-window-scale=<yes|默认no>`**
（推荐设置为no）是否执行HIDPI缩放
默认值情况下初始化的视频窗口随DPI改变大小，例如在缩放200%的4k的显示器打开800p视频后自动执行2倍缩放变1600p。
此外可能还会影响其它脚本部分与此对应属性关联的内容

- **`--window-dragging=<默认yes|no>`**
在画面内按住鼠标左键可拖动窗口？

**Comment by hooke007** [comment]
备楼

**Comment by zhongfly** [comment]
如何设置能让窗口大小与视频原始分辨率一致？也就是不缩放视频画面

**Reply by hooke007** [reply]
无参下只需要加一条 `hidpi-window-scale = no` 就是源视频分辨率。

懒人包的预设里只有这两条有影响
```
autofit-larger = 80%x80%
autofit-smaller = 40%x40%
```

额外的限制来自windows自身 https://github.com/hooke007/mpv_PlayKit/discussions/119

**Comment by nvssdvk** [comment]
问题描述： 使用`window-maximized = yes`设置窗口最大化后，第一个视频窗口最大化正常。 当使用快捷键切换播放列表中的视频时，下一个视频不会最大化，维持原始比例。

请问如何设置可以让播放列表中的所有视频均最大化？

**Reply by hooke007** [reply]
edit
https://mpv.io/manual/master/#options-auto-window-resize

**Comment by rong2er** [comment]
怎么设置成只运行一个播放窗口啊？

**Reply by hooke007** [reply]
不支持 https://github.com/mpv-player/mpv/issues/3811

**Reply by hooke007** [reply]
代理模式： https://github.com/hooke007/mpv_PlayKit/discussions/543

**Comment by karlmorghulis** [comment]


**Reply by hooke007** [reply]
我看不懂你的表述，autofit-smaller 已经让idle的尺寸设定到目标值。“播放视频前”是什么意思

**Reply by karlmorghulis** [reply]
1打开mpv，此时窗口很小
2拖拽视频文件进入mpv，开始播放，窗口变大，尺寸合适。

如何让mpv未播放视频时的默认窗口大小，和播放视频时候的默认窗口大小一致，谢谢大佬。

**Reply by hooke007** [reply]
如前所述，autofit-smaller 控制了idle的尺寸。提高此值解决问题1
mpv默认对每个文件播放时执行一次窗口尺寸匹配，修改此项阻止改行为 https://github.com/hooke007/mpv_PlayKit/blob/179f177ef418e8e4f0fec8477b6bd5a581eb9ed6/portable_config/mpv.conf#L77-L78

==================

除此以外还可以直接使用 --geometry ，参见文档

**Reply by karlmorghulis** [reply]
啊原来是这个意思啊，现在明白了。跟着文档慢慢设置好几天了，之前没接触过这种内容，学习起来很吃力。谢谢大佬愿意回答我的笨蛋问题！

**Comment by juliansy** [comment]
能让窗口边框显示文件路径吗

**Reply by hooke007** [reply]
窗口边框是什么东西，title? 如果是的话仔细阅读属性列表

**Reply by juliansy** [reply]
像是这样, 左上角在文件名后面显示路径

![Snipaste_2025-04-25_22-31-30](https://github.com/user-attachments/assets/5b56e7ce-78b4-4938-b893-2787e12d9b1d)

**Reply by hooke007** [reply]
title不限制放几个属性，那就多放一个属性啊

**Reply by juliansy** [reply]
> title不限制放几个属性，那就多放一个属性啊

我原本不知道怎么写, 不过既然你说不限制数量. 根据你写的`title="实时帧率 ${estimated-display-fps}"`

我便写成`title = "${filename} | 尺寸:${file-size} ${width}×${height} | 路径:${working-directory}\"`

结果还不错

![Snipaste_2025-04-25_23-41-53](https://github.com/user-attachments/assets/09d46a80-7db4-473c-bb3b-a8a212b85618)

谢谢指点
